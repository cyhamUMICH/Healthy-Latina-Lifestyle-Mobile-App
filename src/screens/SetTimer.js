import React, { useState, useEffect } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { Icon } from 'react-native-elements';
import { styles } from '../styles/Styles';
import { colors } from '../styles/Colors';

const SetTimer = (props) => {
  const [duration, setDuration] = useState(0);
  const [numberStack, setNumberStack] = useState("000000");

  const calculateDuration = () => {
    let hours = parseInt(numberStack.substr(0, 2));
    let minutes = parseInt(numberStack.substr(2, 2));
    let seconds = parseInt(numberStack.substr(4,2));

    setDuration((60 * 60 * hours) + (60 * minutes) + seconds);
  };

  const addNumber = (num) => {
    if (numberStack[0] == "0") {
      let newNumberStack = numberStack.substr(1,5) + num;
      setNumberStack(newNumberStack);
    }
  };

  const removeNumber = () => {
    let newNumberStack = "0" + numberStack.substr(0,5);
    setNumberStack(newNumberStack);
  };

  useEffect(() => {
    calculateDuration();
  }, [numberStack]);

  return(
    <View style={styles.app}>
      <View style={styles.fullWidthWindow}>
        <View style={styles.timerScreen}>
          <View style={styles.setTimerDisplayContainer}>
            <Text style={styles.setTimerDisplayNumber}>{numberStack.substr(0, 2)}</Text>
            <Text style={styles.setTimerDisplayColon}>:</Text>
            <Text style={styles.setTimerDisplayNumber}>{numberStack.substr(2, 2)}</Text>      
            <Text style={styles.setTimerDisplayColon}>:</Text>
            <Text style={styles.setTimerDisplayNumber}>{numberStack.substr(4, 2)}</Text>        
          </View>
          <View style={styles.setTimerDisplayContainer}>
            <Text style={styles.setTimerDisplayLabel}>Hours</Text>
            <Text style={styles.setTimerDisplayLabelSpace}></Text>
            <Text style={styles.setTimerDisplayLabel}>Minutes</Text>
            <Text style={styles.setTimerDisplayLabelSpace}></Text>
            <Text style={styles.setTimerDisplayLabel}>Seconds</Text>
          </View>

          <View style={styles.setTimerButtonsContainer}>
            <View style={styles.timerButtonsRow}>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("1")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>1</Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("2")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>2</Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("3")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>3</Text>
                </View>
              </TouchableOpacity>
            </View>
            <View style={styles.timerButtonsRow}>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("4")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>4</Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("5")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>5</Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("6")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>6</Text>
                </View>
              </TouchableOpacity>
            </View>
            <View style={styles.timerButtonsRow}>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("7")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>7</Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("8")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>8</Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("9")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>9</Text>
                </View>
              </TouchableOpacity>
            </View>
            <View style={styles.timerButtonsRow}>
              <TouchableOpacity style={styles.setTimerIconButton} 
                onPress={() => {
                  if (duration != 0) {
                    props.navigation.navigate("Timer", {
                      duration: duration,
                      isPlaying: true
                    });
                  }
                }}>
                <View style={styles.floatingActionIcon}>
                  <Icon name="play-arrow" type="material" size={styles.timerIconSize} color={colors.text} />
                </View>
              </TouchableOpacity>
              <TouchableOpacity style={styles.setTimerNumberButton} 
                onPress={() => addNumber("0")}>
                <View style={styles.floatingActionIcon}>
                  <Text style={styles.setTimerNumberButtonText}>0</Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity style={styles.setTimerIconButton} 
                onPress={removeNumber} onLongPress={() => setNumberStack("000000")}>
                <View style={styles.floatingActionIcon}>
                  <Icon name="backspace" type="material" size={styles.timerIconReducedSize} color={colors.text} />
                </View>
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </View>
    </View>
  );
};

export default SetTimer;