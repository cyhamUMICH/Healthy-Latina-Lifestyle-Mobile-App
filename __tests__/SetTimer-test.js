// Code from: https://github.com/byCedric/expo-guide-ci/blob/master/__tests__/App-test.js
//      and https://jestjs.io/docs/tutorial-react-native
import React from 'react';
import SetTimer from '../src/screens/SetTimer';
import renderer from 'react-test-renderer';

test('Renders Set Timer With the Loading Screen', async () => {
  const tree = renderer.create(<SetTimer />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders Set Timer Without the Loading Screen', async () => {
  const tree = renderer.create(<SetTimer skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});