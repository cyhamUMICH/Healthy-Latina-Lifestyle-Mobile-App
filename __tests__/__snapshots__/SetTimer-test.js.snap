// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Set Timer With the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": "2%",
        "paddingTop": "2%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "marginTop": "10%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginHorizontal": "10%",
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          00
        </Text>
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          :
        </Text>
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          00
        </Text>
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          :
        </Text>
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          00
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginHorizontal": "10%",
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Hours
        </Text>
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        />
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Minutes
        </Text>
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        />
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Seconds
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": "10%",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginTop": "10%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                1
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                2
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                3
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginTop": "10%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                4
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                5
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                6
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginTop": "10%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                7
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                8
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                9
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginTop": "10%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#d3d3d3",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                0
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#d3d3d3",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Renders Set Timer Without the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": "2%",
        "paddingTop": "2%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "marginTop": "10%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginHorizontal": "10%",
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          00
        </Text>
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          :
        </Text>
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          00
        </Text>
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          :
        </Text>
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          00
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginHorizontal": "10%",
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Hours
        </Text>
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        />
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Minutes
        </Text>
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        />
        <Text
          style={
            Object {
              "flex": 2,
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Seconds
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": "10%",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginTop": "10%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                1
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                2
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                3
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginTop": "10%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                4
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                5
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                6
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginTop": "10%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                7
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                8
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                9
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginTop": "10%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#d3d3d3",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#00A89D",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                0
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#d3d3d3",
                "borderRadius": 667,
                "height": null,
                "marginHorizontal": "5%",
                "opacity": 1,
                "width": "15%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
